agent.go

```diff
@@ -336,25 +340,22 @@ func (c *controller) agentDriverNotify(d driverapi.Driver) {
 		return
 	}
 
-	d.DiscoverNew(discoverapi.NodeDiscovery, discoverapi.NodeDiscoveryData{
-		Address:     agent.advertiseAddr,
+	if err := d.DiscoverNew(discoverapi.NodeDiscovery, discoverapi.NodeDiscoveryData{
+		Address:     agent.dataPathAddr,
```

Comment:
On the same line, the decision of setting `Address: agent.dataPathAddr` or `Address: agent.advertiseAddr` IMO should be taken by libnetwork code based on the fact `if agent.dataPathAddr != ""`